code-collab-app/src/editorpage.js

```diff
@@ -32,70 +30,68 @@ class Editorpage extends React.Component {
   }
 
   // Locks a cell so the user cannot edit it
   // key: the index of the cell to lock
   lockEditorCell = (key) => {
-    let temp = this.state.cellLocked;
-    temp[key] = true;
-    this.setState({ cellLocked:temp });
+    // Lock cell so user may not edit
+    this.props.peer_data.set_cell_locked(key, true);
   }
 
-  // Unlocks a cell so the user may edit it
+  // Unlock a cell so the user may edit it
   // key: the index of the cell to unlock
   unlockEditorCell = (key) => {
-    let temp = this.state.cellLocked;
-    temp[key] = false;
-    this.setState({ cellLocked:temp });
+    this.props.peer_data.set_cell_locked(key, false);
+
   }
 
   // Requests lock on cell from all peers in session
   // key: index of cell
   requestEditorCellLock = (key) => {
     // TODO - implement socket.io request to lock this cell on all other peers in session
+    // Will probably involve calling lockEditorCell()
     console.log('Requesting cell lock from all peers for cell ' + key);
     return true;
   }
 
   // Broadcast cell update and release lock for all other peers in session
   // key: index of cell
   // cellContents: text contents of the cell update
-  broadcastCellUpdate = (key, cellContents) => {
+  broadcastCellUpdate = (key) => {
     // TODO - implement socket.io broadcast to update cell for all other peers and release lock
+    // Will probably involve calling unlockEditorCell()
     console.log('Broadcasting cell update to peers for cell ' + key);
   }
 
-  // Event handler for cursor entering a cell. Requests a lock on the cell from
-  // all peers
+  // Event handler for cursor entering a cell. Requests a lock on the cell from all peers.
   // e: onFocus event
   // key: index of cell
   selectCellEvent = (e, key) => {
     e.preventDefault();
-    // BROADCAST UPDATE AND RELEASE LOCK FROM LAST CELL EDITED
-    // Get text of cell just released from editing
-    const lastCellEdited = this.state.cellCurrentlyEditing;
-    if (lastCellEdited !== null) {
-      // Broadcast updates to cell just released from editing
-      let text = this.state.cellText[lastCellEdited]; // Select that cell and extract the text
-      this.broadcastCellUpdate(lastCellEdited, text);
-    }
-
-    // REQUEST LOCK ON NEW CELL
-    this.setState({ cellCurrentlyEditing:null });
+    // Request lock on cell from all peers
     const lockApproved = this.requestEditorCellLock(key);
     if (lockApproved === false) return;
-    // Save the cell currently being edited
-    this.setState({ cellCurrentlyEditing:key });
+    // Remember the cell currently being edited
+    this.props.peer_data.set_current_cell(key);
+  }
+
+  // Event handler for cursor leaving a cell. Broadcasts update for that cell and releases lock.
+  // e: onBlur event
+  // key: index of cell
+  leaveCellEvent = (e, key) => {
+    e.preventDefault();
+    // Get index of last cell edited
+    const lastCellEdited = this.state.cellCurrentlyEditing;
```

Comment:
When I tested this branch it seems the `cellCurrentlyEditing` and `cellCount` state variables never gets updated in the editorpage.

Testing indicates `this.props.peer_data.set_current_cell(key);` is called, but maybe this doesn't make it down to editorpage somehow?

Reply:
Further testing indicates `cellText` and `cellLocked` state variables seem to update fine and trickle down to the editorpage

Reply:
Yeah, I am not quite sure. I think it is because those variables not really used anywhere in rendering. So to get around this issue, I am exposing methods to get the state of `cellCurrentlyEditing` and `cellCount` directly from the peer_data store, and remove them as state variables in Editor page. Thoughts?

Reply:
@BKDuncan this might be right up your alley. Thoughts on best practice? Kinda seems like the editorpage state vars aren't properly referencing the peer_data ones.

Reply:
Also, I don't seem to find the usage of cellCount anywhere. So I am going to remove it from the editor page for now and we can add a method in the peer_data store if we need it later.

Reply:
Good catch, I think `cellCount` was a remnant from a previous architecture of rendering the cells.

Reply:
This is fixed in the latest commit.